feat(core): desugar [()] to [prop] and (prop-change)

vsavkin opened the pull request:
BREAKING CHANGE

Before `<cmp [(prop)]="field"> was desugared to <cmp [prop]="field" (prop)="field=$event">`
After `<cmp [(prop)]="field"> was desugared to <cmp [prop]="field" (prop-change)="field=$event">`

mhevery commented:
For `upgrade`

1)

pkozlowski-opensource commented:
Could you add backticks around code samples in the commit message so it shows correctly in the changelog?

mhevery commented:
Also I think you need to rebase.

vsavkin committed:
feat(core): desugar [()] to [prop] and (prop-change)

BREAKING CHANGE

Before

```
<cmp [(prop)]="field"> was desugared to <cmp [prop]="field" (prop)="field=$event">
```

After
```
<cmp [(prop)]="field"> is desugared to <cmp [prop]="field" (prop-change)="field=$event">
```

mary-poppins commented:
Merging PR #4658 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-4658](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-4658).

rkj commented:
Could you provide any rationale why this was changed? Should all events be named with `-change` suffix moving on?

PascalPrecht commented:
No, not all events have to have `-change` suffix. Only the ones that can be
used with 2-way binding syntax.

I think the reason for this change was just for semantic reasons. Before
this change, an event that is being used with 2-way syntax would be named
like the property. Having -change as suffix makes it semantically an
action. So this makes sense.

On Tue, Nov 24, 2015, 7:49 PM Roman <notifications@github.com> wrote:

> Could you provide any rationale why this was changed? Should all events be
> named with -change suffix moving on?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/angular/pull/4658#issuecomment-159369818>.
>

rkj commented:
Well, we still have events named like properties, just we need to change them depending if we want to use 2-way binding sugar. Thanks for comments!
